packages/dci-api/src/async-search/async-search-handler.ts

```diff
@@ -54,6 +60,29 @@ export async function asyncSearchHandler(
     throw new ValidationError(fromZodError(result.error).message)
   }
   const payload = result.data
+
+  if (payload.signature !== undefined) {
+    try {
+      const signedPayload = await verifySignature(
+        payload.signature,
+        `${payload.header.sender_id}/.well-known/jwks.json`
+      )
+      const parsedSignedPayload =
+        asyncSearchRequestSchema.safeParse(signedPayload)
```

Comment:
I had to parse the signedPayload as we do some transforms when validating with zod otherwise the payloads won't match